packages/main/src/MessageView.ts

```diff
@@ -0,0 +1,246 @@
+})
+/**
+ * Fired when an item is being clicked.
+ *
+ * @event sap.ui.webc.main.MessageView#item-click
+ * @public
+ */
+@event("item-click", {
+	detail: {
+		item: { type: HTMLElement },
+	},
+})
+class MessageView extends UI5Element {
+	/**
+	 * Defines the items of the component.
+	 *
+	 * @type {sap.ui.webc.main.MessageViewItem[]}
+	 * @name sap.ui.webc.main.MessageView.prototype.default
+	 * @slot items
+	 * @public
+	 */
+	@slot({ type: HTMLElement, "default": true })
+	items!: Array<MessageViewItem>;
+
+	/**
+	 * Stores the selected messge type
+	 * @type {string}
+	 * @private
+	 */
+	@property()
+	_selectedMessageType!: string;
+
+	/**
+	 * Stores the list of messages
+	 * @type {Array<MessageItem>}
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	_messages!: Array<MessageItem>
+
+	/**
+	 * Stores the selected message position if any
+	 * @type {Integer}
+	 * @private
+	 */
+	@property({ type: Integer })
+	_selctedMessagePosition!: number
+
+	static i18nBundle: I18nBundle;
+
+	static async onDefine() {
+		MessageView.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	constructor() {
+		super();
+		this._selctedMessagePosition = 0;
+		this._messages = [];
+	}
+
+	onBeforeRendering() {
+		// Enhance the message-view-items with proper message icon and design classes
+		this._messages = this.items.map((item, index) => {
+			return {
+				message: item,
+				messageIcon: MessageTypeIcon[item.type as keyof typeof MessageTypeIcon],
+				messageDesign: DesignClassesMapping[item.type as keyof typeof DesignClassesMapping],
+				visible: true,
+				position: index + 1,
+				listItemType: item.description ? ListItemType.Navigation : ListItemType.Inactive,
+			};
+		});
+
+		if (this._messages.length === 1) {
+			this._selctedMessagePosition = 1;
+		}
+	}
+
+	closeSelectedMessage() {
+		if (this.hasSelectedMessage) {
+			this._selectedMessage.message.onViewChange(MessageViewMode.List);
+		}
+		this._selctedMessagePosition = 0;
+	}
+
+	_filterByMessageType(e: CustomEvent<SegmentedButtonSelectionChangeEventDetail>) {
+		if (e.detail.selectedItems && e.detail.selectedItems.length) {
+			this._selectedMessageType = e.detail.selectedItems[0].id;
+		}
+	}
+
+	_selectListItem(e: any) {
+		const item = e.detail.item;
+		this.fireEvent("item-click", { item });
+
+		this._selctedMessagePosition = item.position ? item.position : 0;
+
+		if (this._selectedMessage) {
+			this._selectedMessage.message.onViewChange(MessageViewMode.Details);
+		}
+	}
+
+	get messageItemsByType() {
+		const result: { messageType: string, messageCount: number, messageIcon: string, messageDesign: string, pressed: boolean }[] = [];
+
+		Object.values(MessageType).forEach(mt => {
+			const count = this._messages.filter(i => i.message.type === mt).length;
+			if (count > 0) {
+				result.push({
+					messageType: mt,
+					messageCount: count,
+					messageIcon: MessageTypeIcon[mt as keyof typeof MessageTypeIcon],
```

Comment:
Here is incorrect to cast `mt` to MessageTypeIcon / DesignClassesMapping as we are not iterating over these enums.